.eslintrc.json

```diff
@@ -0,0 +1,343 @@
+      "error",
+      "never"
+    ],
+    "array-bracket-spacing": [
+      "error",
+      "never"
+    ],
+    "object-curly-spacing": [
+      "error",
+      "never"
+    ],
+    "space-before-blocks": "error",
+    "space-infix-ops": "error",
+    "space-unary-ops": "error",
+    "max-len": [
+      "error",
+      120,
+      2,
+      {
+        "ignoreUrls": true,
+        "ignorePattern": "\\/\\^|import"
+      }
+    ],
+    "quotes": [
+      "error",
+      "single"
+    ],
+    "quote-props": [
+      "error",
+      "consistent-as-needed"
+    ],
+    "curly": [
+      "error",
+      "all"
+    ],
+    "eol-last": "error",
+    "no-with": "error",
+    "no-var": "error",
+    "prefer-arrow-callback": "error",
+    "arrow-body-style": [
+      "error",
+      "as-needed"
+    ],
+    "object-shorthand": [
+      "error",
+      "always"
+    ],
+    "prefer-const": "error",
+    "camelcase": "error",
+    "comma-dangle": "error",
+    "dot-notation": "error",
+    "getter-return": "error",
+    "id-match": "error",
+    "new-parens": "error",
+    "no-caller": "error",
+    "no-cond-assign": "off",
+    "no-empty": "error",
+    "no-eval": "error",
+    "no-fallthrough": "error",
+    "no-new-wrappers": "error",
+    "no-shadow": [
+      "error",
+      {
+        "hoist": "all",
+        "builtinGlobals": false
+      }
+    ],
+    "no-throw-literal": "error",
+    "no-unused-expressions": "error",
+    "one-var-declaration-per-line": [
+      "error",
+      "initializations"
+    ],
+    "radix": "off",
+    "space-before-function-paren": [
+      "error",
+      {
+        "anonymous": "never",
+        "asyncArrow": "always",
+        "named": "never"
+      }
+    ],
+    "spaced-comment": "error",
+    "valid-typeof": "off",
+    "yoda": "off",
+    "jsx-quotes": [
+      "error",
+      "prefer-single"
+    ],
+    "no-invalid-this": "off", // see typescript-eslint issue 491
+    "no-dupe-class-members": "off", // allow overload syntax
+    "@typescript-eslint/explicit-member-accessibility": [
+      "error",
+      {
+        "accessibility": "no-public"
+      }
+    ],
+    "@typescript-eslint/adjacent-overload-signatures": "error",
+    "@typescript-eslint/array-type": [
+      "error",
+      {
+        "default": "array-simple"
+      }
+    ],
+    "@typescript-eslint/await-thenable": "error",
+    "@typescript-eslint/ban-types": "error",
+    "@typescript-eslint/class-name-casing": "error",
+    "@typescript-eslint/consistent-type-assertions": "error",
+    "@typescript-eslint/consistent-type-definitions": "off",
+    "@typescript-eslint/no-unused-vars": "error",
+    "@typescript-eslint/interface-name-prefix": "off",
+    "@typescript-eslint/member-delimiter-style": [
+      "error",
+      {
+        "multiline": {
+          "delimiter": "semi",
+          "requireLast": true
+        },
+        "singleline": {
+          "delimiter": "semi",
+          "requireLast": true
+        },
+        "overrides": {
```

Comment:
Remove overrides for tests since we don't have any

Reply:
Also, I don't think this is a valid eslintrc with "overrides" incorrectly placed. Please run the linter to see it works as expectd before commiting changes.